src/encord_active/lib/common/data_utils.py

```diff
@@ -146,10 +147,42 @@ def download_file(
     return destination
 
 
-def download_image(url: str) -> Image.Image:
+def url_to_file_path(url: str, project_dir: Path) -> Optional[Path]:
+    if url.startswith("/") or url.startswith("./"):
+        return Path(url)
     if url.startswith("file:"):
-        image_path = Path(unquote(urlparse(url).path))
-        return Image.open(image_path)
+        return Path(unquote(urlparse(url).path))
+    if url.startswith("relative://"):
+        relative_path = url[len("relative://"):]
+        return project_dir / Path(relative_path)
+    if url.startswith("absolute://"):
+        absolute_path = url[len("absolute:/"):]
+        return Path(absolute_path)
+    return None
+
+
+def file_path_to_url(path: Path, project_dir: Path, relative: Optional[bool] = None) -> str:
+    path = path.absolute().resolve()
```

Comment:
If `path` starts with `~/` this statement will yield a bad results, we should use `expanduser` instead of `absolute`.
<img width="460" alt="image" src="https://github.com/encord-team/encord-active/assets/115489098/b33e5e54-0698-4bee-8b44-4b7d167faa8f">

```suggestion
    path = path.expanduser().resolve()
```